apps/vc-api/src/config/db.ts

```diff
@@ -0,0 +1,86 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { TypeOrmModule, TypeOrmModuleOptions } from '@nestjs/typeorm';
+import { ConfigService } from '@nestjs/config';
+import { DB_TYPES } from './env-vars-validation-schema';
+import * as path from 'path';
+
+/**
+ * Inspired by https://dev.to/webeleon/unit-testing-nestjs-with-typeorm-in-memory-l6m
+ */
+const inMemoryDBConfig: TypeOrmModuleOptions = {
+  type: 'better-sqlite3',
+  database: ':memory:',
+  dropSchema: true,
+  synchronize: true,
+  keepConnectionAlive: true // https://github.com/nestjs/typeorm/issues/61
+};
+
+const commonOptions = {
+  autoLoadEntities: true // https://docs.nestjs.com/techniques/database#auto-load-entities
+};
+
+export const typeOrmConfigFactory = (config: ConfigService): TypeOrmModuleOptions => {
+  if (config.get<DB_TYPES>('DB_TYPE') === DB_TYPES.SQLITE_INMEMORY) {
+    return {
+      ...inMemoryDBConfig,
+      ...commonOptions
+    };
+  }
+
+  if (config.get<DB_TYPES>('DB_TYPE') === DB_TYPES.SQLITE) {
+    const SQLITE_FILE = config.get('SQLITE_FILE');
+    if (!SQLITE_FILE) {
+      throw new Error(`SQLITE_FILE not set`);
+    }
+
+    // this is required because migration generation is executed from the migrations subfolder
+    const databaseFilePath = path.isAbsolute(SQLITE_FILE)
+      ? SQLITE_FILE
+      : path.resolve(__dirname, '../..', SQLITE_FILE);
+
+    return {
+      type: 'better-sqlite3',
+      database: databaseFilePath,
+      dropSchema: config.get('DB_DROP_SCHEMA'),
+      synchronize: config.get('DB_SYNCHRONIZE'),
+      migrationsRun: config.get('DB_RUN_MIGRATIONS'),
+      migrations: [`${path.resolve(__dirname, '../migrations/sqlite')}/*.{ts,js}`],
+      ...commonOptions
+    };
+  }
+
+  if (config.get<DB_TYPES>('DB_TYPE') === DB_TYPES.POSTGRES) {
+    return {
+      type: 'postgres',
+      host: config.get('POSTGRES_DB_HOST'),
+      port: +config.get('POSTGRES_DB_PORT'),
+      username: config.get('POSTGRES_DB_USER'),
+      password: config.get('POSTGRES_DB_PASSWORD'),
+      database: config.get('POSTGRES_DB_NAME'),
+      dropSchema: config.get('DB_DROP_SCHEMA'),
+      synchronize: config.get('DB_SYNCHRONIZE'),
+      migrationsRun: config.get('DB_RUN_MIGRATIONS'),
+      migrations: [`${path.resolve(__dirname, '../migrations/pg')}/*.{ts,js}`],
+      ...commonOptions
+    };
+  }
+};
+
+export const typeOrmInMemoryModuleFactory = () =>
```

Comment:
We can remove this now, since sqlite config is returned by `typeOrmConfigFactory`

Reply:
It is still used in tests and only there as they are executed with an in-memory database.

Reply:
Hmm, why tests can't use `typeOrmConfigFactory(config of sqlite datasource)` ?

Reply:
It was just simpler for me to keep this as a wrapper than mocking `ConfigService` and injecting it in tests. Do you think it is worth the effort to do this?

Reply:
it would look like this, what do you think?
```typescript
        TypeOrmModule.forRoot(
          typeOrmConfigFactory({
            get: (key): Record<string, unknown> => {
              return { DB_TYPE: DB_TYPES.SQLITE_IN_MEMORY }[key];
            }
          } as unknown as ConfigService)
        ),
```
instead of this:
```typescript
typeOrmInMemoryModuleFactory(),
```